src/lib/shady-render.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {render as baseRender, Template, templateCaches, TemplateResult} from '../lit-html.js';
+
+export {html, svg, TemplateResult} from '../lit-html.js';
+
+declare global {
+  interface Window {
+    ShadyCSS: any;
+  }
+}
+
+const shadyTemplateFactory = (scopeName: string) =>
+    (result: TemplateResult) => {
+      const cacheKey = `${result.type}--${scopeName}`;
+      let templateCache = templateCaches.get(cacheKey);
+      if (templateCache === undefined) {
+        templateCache = new Map<TemplateStringsArray, Template>();
+        templateCaches.set(cacheKey, templateCache);
+      }
+      let template = templateCache.get(result.strings);
+      if (template === undefined) {
+        const element = result.getTemplateElement();
+
+        if (typeof window.ShadyCSS === 'object') {
+          window.ShadyCSS.prepareTemplate(element, scopeName);
+        }
+
+        template = new Template(result, element);
+        templateCache.set(result.strings, template);
+      }
+      return template;
+    };
+
+export function render(
+    result: TemplateResult,
+    container: Element|DocumentFragment,
+    scopeName: string) {
+  return baseRender(result, container, shadyTemplateFactory(scopeName));
```

Comment:
How does this play with lit-extended? Seems like with this you can choose between either lit-extended or shady dom..?

Reply:
lit-extended now creates and exports its own `html` and `svg` template literals instead of its own `render` function, so this should work fine.

Example usage:

```js
import {html} from 'path/to/lit-html/lit-html.js';
import {render} from 'path/to/lit-html/lib/shady-render.js';

render(html`<div>test</div>`, container, 'custom-element-name');
```

Reply:
Ahh great! I always thought having the instantiation (and thus also the template syntax) depend on the render and not the template function was an API wart. Good thing thats gone now!

Reply:
Yeah, me too. Not sure why I didn't do it this way originally. 🤷‍♂️

@bgotink this will require some changes for you, but should be straight-forward. WDYT?

Reply:
Should be straightforward enough, similar to the changes made to `lit-extended.js`. I haven't tried it yet though.

Reply:
that sure seems like a better approach :) looking forward to it :)